test/SlowTests/Issues/RavenDB_20059.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Threading.Tasks;
+using SlowTests.Server.Documents.ETL;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_20059 : EtlTestBase
+    {
+        public RavenDB_20059(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void CanModifyGlobalObjectUsingEtlTransformScriptWithDeleteDocumentsBehaviorFunction()
+        {
+            using (var srcStore = GetDocumentStore())
+            using (var destStore = GetDocumentStore())
+            {
+                AddEtl(srcStore, destStore, "Contracts", script:
+@"this.Contact.AdditionalInfo = 13;	
+loadToContractsTemp(this);
+function deleteDocumentsOfContractsBehavior(docId) {
+    return false;
+    }");
+
+                var etlDone = WaitForEtl(srcStore, (_, s) => s.LoadSuccesses > 0);
+
+                using (var session = srcStore.OpenSession())
+                {
+                    session.Store(new Contract { Contact = new Contact { AdditionalInfo = 10 } });
+
+                    session.SaveChanges();
+                }
+                etlDone.Wait(TimeSpan.FromMinutes(1));
+
+                using (var session = destStore.OpenSession())
+                {
+                    var contract = session.Load<Contract>("contracts/1-A/ContractsTemp/0000000000000000001-A");
+                    Assert.NotNull(contract);
+                    Assert.Equal(13, contract.Contact.AdditionalInfo);
+                }
+
+                using (var session = srcStore.OpenSession())
+                {
+                    session.Delete("contracts/1-A");
+                    session.SaveChanges();
+                }
+
+                etlDone = WaitForEtl(srcStore, (_, s) => s.LoadSuccesses > 0);
+                etlDone.Wait(TimeSpan.FromMinutes(1));
+
+                using (var session = destStore.OpenSession())
+                {
+                    var contract = session.Load<Contract>("contracts/1-A/ContractsTemp/0000000000000000001-A");
+
+                    Assert.NotNull(contract);
+                    Assert.Equal(13, contract.Contact.AdditionalInfo);
+                }
+            }
+        }
```

Comment:
Since we ignore the validation errors now, I wonder if we shouldn't add a test which will verify that we get the errors in runtime if the behavior function is faulty. In order words, can we add the test where the behavior function has invalid syntax and assure that we get the errors when the ETL process runs it?

Reply:
Can it be done? Here you can find an utility method that will let you get the transformation alert:

https://github.com/ravendb/ravendb/blob/372739ca538938c8306566c55a342304e4d26239/test/SlowTests/Server/Documents/ETL/EtlTestBase.cs#L224-L254

Reply:
Okay